test/solc/CommandLineParser.cpp

```diff
@@ -426,6 +426,58 @@ BOOST_AUTO_TEST_CASE(invalid_options_input_modes_combinations)
 		}
 }
 
+BOOST_AUTO_TEST_CASE(default_optimiser_sequence)
+{
+	auto const& commandLineOptions = parseCommandLine({"solc", "contract.sol", "--optimize"});
+	BOOST_CHECK_EQUAL(commandLineOptions.optimiserSettings().yulOptimiserSteps, OptimiserSettings::DefaultYulOptimiserSteps);
+	BOOST_CHECK_EQUAL(commandLineOptions.optimiserSettings().yulOptimiserCleanupSteps, OptimiserSettings::DefaultYulOptimiserCleanupSteps);
+}
+
+BOOST_AUTO_TEST_CASE(valid_optimiser_sequences)
+{
+	vector<string> validSequenceInputs {
+		":", 						// Empty optimizaiton sequence and empty cleanup sequence
+		":fDn", 					// Empty optimization sequence and specified cleanup sequence
+		"dhfoDgvulfnTUtnIf:", 		// Specified optimizaiton sequence and empty cleanup sequence
+		"dhfoDgvulfnTUtnIf:fDn", 	// Specified optimization sequence and cleanup sequence
+		"dhfo[Dgvulfn]TUtnIf:f[D]n" // Specified and nested optimization and cleanup sequence
+	};
+
+	vector<tuple<string, string>> expectedParsedSequences {
+		{ "", "" },
+		{ "", "fDn"},
+		{ "dhfoDgvulfnTUtnIf", ""},
+		{ "dhfoDgvulfnTUtnIf", "fDn"},
+		{ "dhfo[Dgvulfn]TUtnIf", "f[D]n"}
+	};
+
+	BOOST_CHECK_EQUAL(validSequenceInputs.size(), expectedParsedSequences.size());
+
+	for (size_t i = 0; i < validSequenceInputs.size(); ++i)
```

Comment:
Why not a range loop? Does not look like you're doing anything non-trivial with the loop counter.

Reply:
```
auto const& [expectedYulOptimiserSteps, expectedYulCleanupSteps] = expectedParsedSequences[i];
```
I'm using it to fetch a value at the same index from another array.

Reply:
Ah, ok, my bad. Was only skimming through it and did not notice.